elasticsearch.md

```diff
@@ -1487,63 +1488,166 @@ ReadonlyREST can write events very similarly to Logstash into to a series of ind
+
+```text
+readonlyrest:
+
+  audit:
+    enabled: true
+    outputs: [ index, log ]
+
+    ...
+```
+
+In case when you want to disable given output, add the `enabled: false` to the output config:
+```text
+readonlyrest:
+  audit:
+    enabled: true
+    outputs: 
+    - type: index
+    - type: log
+      enabled: false # by default is true
+    ...
+```
+
+#### The 'index' output specif configurations
+
+##### Custom audit indices name and time granularity
+
+It is possible to change the name of the produced audit indices by specifying a template value as `index_template`.
+
+Example: tell ROR to write on the monthly index.
+
+```text
+readonlyrest:
+  audit:
+    enabled: true
+    outputs: 
+    - type: index
+      index_template: "'custom-prefix'-yyyy-MM"  # <--monthly pattern
+  ...
+```
+**⚠️IMPORTANT**: notice the single quotes inside the double quoted expression. This is the same syntax used for [Java's SimpleDateFormat](https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html).
+
+##### Custom audit cluster
 
-If you want to log the request content then an additional serializer is provided. This will log the entire user request within the content field of the audit event. To enable, configure the audit\_serializer parameter as below.
+It's possible to set a custom audit cluster responsible for audit events storage. When a custom cluster is specified, items will be sent to defined cluster nodes instead of the local one.
 
 ```text
 readonlyrest:
   audit:
-    collector: true
-    serializer: tech.beshu.ror.requestcontext.QueryAuditLogSerializer
+    enabled: true
+    outputs: 
+    - type: index
+      cluster: ["https://user1:password@auditNode1:9200", "https://user2:password@auditNode2:9200"]
   ...
 ```
 
-### Custom audit indices name and time granularity
+Setting `audit.cluster` is optional, it accepts non empty list of audit cluster nodes URIs.
 
-It is possible to change the name of the produced audit log indices by specifying a template value as `audit.index_template`.
+#### The 'log' output specif configurations
 
-Example: tell ROR to write on monthly index.
+The `log` output uses a dedicated logger to write the audit events in files. The logger applies the `INFO` log level to the appended audit events.
 
+To configure audit events writing to logs, you need to add to the `readonlyrest` configuration the following settings:
 ```text
 readonlyrest:
   audit:
-    collector: true
-    index_template: "'custom-prefix'-yyyy-MM"  # <--monthly pattern
+    enabled: true
+    outputs: 
+    - type: log
   ...
 ```
 
-**⚠️IMPORTANT**: notice the single quotes inside the double quoted expression. This is the same syntax used for [Java's SimpleDateFormat](https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html).
+In the next step you need to configure the logger for audit events in the `$ES_PATH_CONF/config/log4j2.properties`. Here is an example config with default logger name:
```

Comment:
Is it needed? I thought it's an optional part (when eg. you want to have a separate file with audit logs)

Reply:
Yes, it's optional. I think I was focused too much on custom logging. 
This is how the log looks like without additional changes in log4j properties:
```[2023-04-04T20:14:34,415][INFO ][readonlyrest_audit       ] {"error_message":"error","headers":[],"acl_history":"","origin":"localhost","match":false,"final_state":"ERRORED","destination":"localhost","task_id":0,"type":"default-type","req_method":"GET","path":"_search","indices":[],"@timestamp":"2018-12-31T23:01:59Z","content_len_kb":0,"error_type":"Exception","correlation_id":"7029bd54-2e0c-4e10-a137-933f15efb52e","processingMillis":134334755414,"action":"default-action","block":"error","id":"6ff2b948-d509-4bf1-be3f-4bcda0b4f0fb","content_len":0}```

Reply:
It's ok, but we should clearly mention that I user wants to have the audit in separate file, they can do it like that